src/main/kotlin/org/elm/ide/docs/ElmDocumentationProvider.kt

```diff
@@ -157,6 +159,17 @@ private fun documentationFor(clause: ElmAsClause): String? = buildString {
     return documentationFor(decl)
 }
 
+private fun StringBuilder.renderDefinitionLocation(element: ElmPsiElement) {
+    val module = element.elmFile.getModuleDecl() ?: return
+    renderDefinedInQIDs(module.upperCaseQID.upperCaseIdentifierList)
+}
+
+private fun StringBuilder.renderDefinedInQIDs(ids: List<PsiElement>) {
```

Comment:
Assuming the comment on 153 is applied, then instead of using `joinTo()` to rebuild the dotted string, I'd just use `getText()` on the QID itself.

`getText()` should also be used on line 174 (`ref.upperCaseQID.upperCaseIdentifierList.joinToString(".") { it.text }`) rather than rebuilding the dotted string.